File: backend/src/main/java/com/moa/constant/ClubApplicationStatus.java
package com.moa.constant;

/**
 * club_applications.status 컬럼과 매핑되는 동아리 가입 신청 상태.
 *
 * PENDING으로 생성되며, 동아리장이 승인/거절하면 APPROVED/REJECTED로 바뀐다.
 * TODO(leader-approval): 아직 APPROVED/REJECTED로 바꿔주는 승인/거절
 * 엔드포인트가 없다(동아리장 기능은 이번 범위 밖). 지금은 신청서를 내면
 * PENDING 상태로 남아있고, REJECTED일 때만 재신청(resubmit)이 가능하다.
 */
public enum ClubApplicationStatus {
    PENDING,
    APPROVED,
    REJECTED
}

File: frontend/lib/features/club/models/club_application_status.dart
/// backend ClubDetailResponse.applicationStatus(ClubApplicationStatus)와 매핑되는
/// 동아리 가입 신청 상태. joined가 true면 이 값은 항상 none이다.
enum ClubApplicationStatus {
  /// 아직 신청서를 낸 적 없음(서버 응답의 applicationStatus == null).
  none,

  /// 신청서를 냈고 동아리장 승인을 기다리는 중.
  pending,

  /// 거절됨. 다시 지원할 수 있다.
  rejected;

  static ClubApplicationStatus fromJson(String? value) {
    switch (value) {
      case 'PENDING':
        return ClubApplicationStatus.pending;
      case 'REJECTED':
        return ClubApplicationStatus.rejected;
      default:
        return ClubApplicationStatus.none;
    }
  }
}
